java/test/jmri/jmrit/display/layoutEditor/LayoutEditorChecksTest.java

```diff
@@ -1,40 +1,349 @@
+                2, resultsPopupMenu.getSubElements().length);
+        JMenuItem resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsSubMenuItem0.getText(): ", myBlockName + ":#1", resultsSubMenuItem.getText());
+        resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsSubMenuItem1.getText(): ", myBlockName + ":#2", resultsSubMenuItem.getText());
+    }   //testCheckNonContiguousBlocks
+
+    @Test
+    public void testCheckUnnecessaryAnchors() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("toolsMenuTitle not null", toolsMenuTitle);
+        JMenuOperator toolsJMO = new JMenuOperator(layoutEditorEFO, toolsMenuTitle);
+        Assert.assertNotNull("CheckMenuTitle not null", checkMenuTitle);
+        Assert.assertNotNull("checkUnnecessaryAnchorsMenuTitle not null", checkUnnecessaryAnchorsMenuTitle);
+
+        JPopupMenu toolsPopupMenu = toolsJMO.getPopupMenu();
+        JMenuItem checkMenuItem = (JMenuItem) toolsPopupMenu.getComponent(0);
+        Assert.assertEquals("checkMenuItem.getText(): ",
+                checkMenuTitle, checkMenuItem.getText());
+
+        JMenuOperator checkJMO = new JMenuOperator((JMenu) checkMenuItem);
+        JPopupMenu checksPopupMenu = checkJMO.getPopupMenu();
+        JMenuItem checkUnnecessaryAnchorsMenuItem = (JMenuItem) checksPopupMenu.getComponent(3);
+        Assert.assertEquals("subMenuItem.getText(): ",
+                checkUnnecessaryAnchorsMenuTitle, checkUnnecessaryAnchorsMenuItem.getText());
+
+        JMenuOperator checkUnnecessaryAnchorsJMO = new JMenuOperator((JMenu) checkUnnecessaryAnchorsMenuItem);
+        checkUnnecessaryAnchorsJMO.doClick();
+
+        JPopupMenu checkUnnecessaryAnchorsPopupMenu = checkUnnecessaryAnchorsJMO.getPopupMenu();
+
+        //verify results
+        Assert.assertEquals("Correct number of check unnecessary anchors results menu items",
+                2, checkUnnecessaryAnchorsPopupMenu.getSubElements().length);
+        JMenuItem resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsMenuItem0.getText(): ", a1.getName(), resultsMenuItem.getText());
+        resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsMenuItem1.getText(): ", a2.getName(), resultsMenuItem.getText());
+    }   //testCheckUnnecessaryAnchors
+
+    @BeforeClass
+    public static void setUpClass() {
         JUnitUtil.setUp();
-        JUnitUtil.resetProfileManager();
+        if (!GraphicsEnvironment.isHeadless()) {
+            JUnitUtil.resetProfileManager();
+
+            // set default string matching comparator to one that exactly matches and is case sensitive
+            Operator.setDefaultStringComparator(new Operator.DefaultStringComparator(true, true));
```

Comment:
It is entirely possible that this operation on the operator has global implications.  This actually MIGHT be causing failures in tests outside this test class.

Make sure you reset this to the previous default when cleanup occurs.

Reply:
That was my thought as well… (long after the fact). I've added the restore code to most of my PR's that change it (excluding the one that you caught). (IMHO: true, true should be the default… why would you want sloppy matching?!? My initial work-around (before I found this setting) was I had to hard-code indexes into possible matches. Setting this true, true eliminated the hard coded indexes… which IMHO: is a win.)